packages/imt.sol/contracts/internal/InternalLeanIMT.sol

```diff
@@ -67,7 +73,7 @@ library InternalLeanIMT {
 
         self.size += 1;
 
-        self.sideNodes[self.depth] = node;
+        self.sideNodes[treeDepth] = node;
         self.leaves[leaf] = self.size;
```

Comment:
after this change: https://github.com/privacy-scaling-explorations/zk-kit/pull/221/files?diff=unified&w=1#r1532333264, you can do:

```solidity
self.leaves[leaf] = treeSize;
```